src/main/java/org/kohsuke/github/extras/auth/OrgInstallationCredentialProvider.java

```diff
@@ -0,0 +1,70 @@
+package org.kohsuke.github.extras.auth;
+
+import org.kohsuke.github.*;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Date;
+
+/**
+ * This helper class provides an example on how to authenticate a GitHub instance with an installation token, that will
+ * be automatically refreshed when required.
+ */
+public class OrgInstallationCredentialProvider implements CredentialProvider {
+
+    private final GitHub gitHub;
+
+    private final String organizationName;
+
+    private String latestToken;
+
+    private Date validUntil;
+
+    public OrgInstallationCredentialProvider(String organizationName, GitHub gitHub) {
```

Comment:
Having this take a `GitHub` at construction time makes it more difficult to work with, but it is probably something we can tweak/improve later.

Reply:
I will try to give it a go following your suggestions, thanks a lot!!!!